src/runtime/math.py

```diff
@@ -709,3 +709,64 @@ def remainder(x: f64, y: f64) -> f64:
     if x - y*f64(q) > y*f64(q + i64(1)) - x:
         return x - y*f64(q + i64(1))
     return x - y*f64(q)
+
+
+def sumprod(x: list[f64], y: list[f64]) -> f64:
+    '''
+    Return the sum of products of values from the lists x and y.
+    '''
+    if len(x) != len(y):
+        raise ValueError("The two lists must have the same length.")
+    
```

Comment:
Please add the following line:
```python
i: i32
```

Reply:
I noticed it right after I made the pr thanks for your help!